docs/source/satkill_explanation.rst

```diff
@@ -0,0 +1,240 @@
+        cut_type           : Cut mode to the deconvolution output (`abs` or `rel`) using e_cut
+                            `abs`: cut on the absolute value of the hits.
+                            `rel`: cut on the relative value (to the max) of the hits.
+
+        Returns
+        -------
+        array : Boolean mask of all labelled satellite deposits, True where satellites should be removed.
+
+        References
+        ----------
+        .. [1] https://github.com/scikit-image/scikit-image/blob/main/skimage/morphology/misc.py#L59-L151
+        '''
+        if cut_type is CutType.rel:
+            im_deconv = im_deconv / im_deconv.max()
+
+        # separate different regions below and above e_cut
+        # then label regions (components) appropriately and determine their sizes.
+        labels, component_sizes = collect_component_sizes(im_deconv >= e_cut)
+
+        # check if no satellites within deposit return False array
+        # (mask that removes no satellites).
+        if len(component_sizes) <= 2:
+            return np.full(im_deconv.shape, False)
+
+        # Find regions smaller than `satellite_max_size` and mask them, 
+        # ignoring the first region (background). Read gist for full explanation.
+        too_small      = component_sizes < satellite_max_size
+        too_small[0]   = False 
+        too_small_mask = too_small[labels]
+        return too_small_mask
+
+This mask is then used to remove satellites from 2D deconvolution arrays. Lets step through the logic!
+
+**Stepping through the satellite killing process**
+
+We start in ``generate_satellite_mask()``:
+
+
+.. highlight:: python 
+.. code-block:: python
+
+    if cut_type is CutType.rel:
+            im_deconv = im_deconv / im_deconv.max()
+
+This is a simple check for the ``cut_type``, if it is absolute (``abs``) no changes are needed, the relative (``rel``) changes are self explanatory.
+
+
+.. highlight:: python 
+.. code-block:: python
+
+    labels, component_sizes = collect_component_sizes(im_deconv >= e_cut)
+
+
+As shown above, the function then uses ``collect_component_sizes()`` to collect an array of all the labelled 
+regions, and the sizes of each of those regions. A boolean cut is applied in the argument (``im_deconv >= e_cut``) 
+to ensure the array exists with only 0s and 1s, with 0s being energies below the `e_cut` and 1s being above.
+
+So the array should look something like this (as an example):
+
+
+.. highlight:: python 
+.. code-block:: python
+
+    >>> print(im_mask)
+    array([[1., 1., 0., 0., 1.],
+           [1., 1., 0., 0., 0.],
+           [0., 0., 0., 0., 0.],
+           [0., 0., 1., 1., 1.],
+           [0., 0., 1., 1., 1.]])
+
+The function ``collect_component_sizes()`` completes the following steps:
+
+* ``footprint = ndi.generate_binary_structure(im_mask.ndim, 2)``
+* This creates an n-dimensional array of ``True`` values that are
+  used to map the connectivity of our 1s in the above array, like such:
```

Comment:
I think you meant to format this in a different way? These bullet points look awkward